quickjs.c

```diff
@@ -40007,10 +40021,29 @@ static JSValue js_iterator_from(JSContext *ctx, JSValue this_val,
     return iter;
 }
 
+static JSValue js_create_iterator_helper(JSContext *ctx, JSValue iterator,
+                                         JSIteratorHelperKindEnum kind, JSValue func, uint32_t limit);
+
 static JSValue js_iterator_proto_drop(JSContext *ctx, JSValue this_val,
                                       int argc, JSValue *argv)
 {
-    return JS_ThrowInternalError(ctx, "TODO implement Iterator.prototype.drop");
+    JSValue v;
+    int32_t limit;
+    if (!JS_IsObject(this_val))
+        return JS_ThrowTypeError(ctx, "Iterator.prototype.drop called on non-object");
+    v = JS_ToNumber(ctx, argv[0]);
+    if (JS_IsException(v))
+        return JS_EXCEPTION;
+    if (JS_VALUE_IS_NAN(v))
+        return JS_ThrowRangeError(ctx, "Must be positive");
+    v = JS_ToIntegerFree(ctx, v);
```

Comment:
I need to fix a couple of edge cases here such as Number.MAX_SAFE_INTEGER or Infinity.